# Transaction Sheet

In [1]:
import pandas as pd
import numpy as np

In [2]:
tran = pd.read_excel("KPMG_VI_New_raw_data_update_final.xlsx", sheet_name= "Transactions",skiprows=1)
tran.head(10)

transaction_id  product_id  customer_id transaction_date  online_order  \
0               1           2         2950       2017-02-25           0.0   
1               2           3         3120       2017-05-21           1.0   
2               3          37          402       2017-10-16           0.0   
3               4          88         3135       2017-08-31           0.0   
4               5          78          787       2017-10-01           1.0   
5               6          25         2339       2017-03-08           1.0   
6               7          22         1542       2017-04-21           1.0   
7               8          15         2459       2017-07-15           0.0   
8               9          67         1305       2017-08-10           0.0   
9              10          12         3262       2017-08-30           1.0   

  order_status           brand product_line product_class product_size  \
0     Approved           Solex     Standard        medium       medium   
1     Approved   Trek Bicycles     Standard        medium        large   
2     Approved      OHM Cycles     Standard           low       medium   
3     Approved  Norco Bicycles     Standard        medium       medium   
4     Approved  Giant Bicycles     Standard        medium        large   
5     Approved  Giant Bicycles         Road        medium       medium   
6     Approved        WeareA2B     Standard        medium       medium   
7     Approved        WeareA2B     Standard        medium       medium   
8     Approved           Solex     Standard        medium        large   
9     Approved        WeareA2B     Standard        medium       medium   

   list_price  standard_cost  product_first_sold_date  
0       71.49          53.62                  41245.0  
1     2091.47         388.92                  41701.0  
2     1793.43         248.82                  36361.0  
3     1198.46         381.10                  36145.0  
4     1765.30         709.48                  42226.0  
5     1538.99         829.65                  39031.0  
6       60.34          45.26                  34165.0  
7     1292.84          13.44                  39915.0  
8     1071.23         380.74                  33455.0  
9     1231.15         161.60                  38216.0

In [8]:
tran.shape

(20000, 13)

### Check Missing Values

In [3]:
missing_values = tran.isnull().sum()
missing_values[missing_values>0]

online_order               360
brand                      197
product_line               197
product_class              197
product_size               197
standard_cost              197
product_first_sold_date    197
dtype: int64

Following volumn has missing values:
- online_order               
- brand                      
- product_line               
- product_class              
- product_size               
- standard_cost              
- product_first_sold_date    

All these Data fields must be completed to be able to drive a better insight from the data. (Completeness)

### Customer_id column

In [6]:
tran[tran['customer_id'] >= 4003]

transaction_id  product_id  customer_id transaction_date  online_order  \
8707             8708           0         5034       2017-10-07           0.0   
16700           16701           0         5034       2017-01-27           0.0   
17468           17469           0         5034       2017-01-03           0.0   

      order_status           brand product_line product_class product_size  \
8707      Approved           Solex         Road        medium       medium   
16700     Approved  Norco Bicycles     Standard        medium       medium   
17468     Approved      OHM Cycles         Road        medium       medium   

       list_price  standard_cost  product_first_sold_date  
8707       416.98     312.735016                  42218.0  
16700      360.40     270.299988                  35667.0  
17468      742.54     667.400024                  33879.0

We see here that customer_id number **5034** made a transaction but it's detail is not in our both (CustomerDemographic and CustomerAddress) sheets/table.

### List_price Column

In [9]:
tran[['list_price']].head()

list_price
0       71.49
1     2091.47
2     1793.43
3     1198.46
4     1765.30

Here no currency sign (dollar, euro etc) is mentioned.

### Standard_cost Column

In [15]:
tran[['standard_cost']].head(20)

standard_cost
0           53.62
1          388.92
2          248.82
3          381.10
4          709.48
5          829.65
6           45.26
7           13.44
8          380.74
9          161.60
10         459.71
11          56.93
12         154.40
13        1479.11
14         709.48
15         388.92
16         818.01
17        1167.18
18          13.44
19         829.65

We can see here data inconsistency, some value has currecy sign in it while some has not. Furthermore some value has 2 decimal value while some has more.

### Product_first_sold_date Column

In [18]:
tran[['product_first_sold_date']].head()

product_first_sold_date
0                  41245.0
1                  41701.0
2                  36361.0
3                  36145.0
4                  42226.0

No information is derived from this data. Not correct date column.